src/Tasks.UnitTests/MSBuild_Tests.cs

```diff
@@ -1341,5 +1342,56 @@ public void TargetNameIsCaseInsensitive()
                 File.Delete(projectFile2);
             }
         }
+
+        [Fact]
+        public void NonexistentTargetSkippedWhenSkipNonexistentTargetsSet()
+        {
+            ObjectModelHelpers.DeleteTempProjectDirectory();
+            ObjectModelHelpers.CreateFileInTempProjectDirectory(
+                "SkipNonexistentTargets.csproj",
+                @"<Project>
+                    <Target Name=`t` >
+	                    <MSBuild Projects=`SkipNonexistentTargets.csproj` Targets=`t_nonexistent;t2` SkipNonexistentTargets=`true` />
+                     </Target>
+                    <Target Name=`t2`> <Message Text=`t2 executing` /> </Target>
+                </Project>
+                ");
+
+            MockLogger logger = ObjectModelHelpers.BuildTempProjectFileExpectSuccess(@"SkipNonexistentTargets.csproj");
+
+            // Target t2 should still execute
+            logger.FullLog.ShouldContain("t2 executing");
+            logger.WarningCount.ShouldBe(0);
+            logger.ErrorCount.ShouldBe(0);
+
+            // t_missing target should be skipped
+            logger.FullLog.ShouldContain("Target \"t_nonexistent\" skipped");
+        }
+
+        [Fact]
+        public void NonexistentTargetFailsAfterSkipped()
+        {
+            ObjectModelHelpers.DeleteTempProjectDirectory();
+            ObjectModelHelpers.CreateFileInTempProjectDirectory(
+                "SkipNonexistentTargets.csproj",
+                @"<Project>
+                    <Target Name=`t` >
+	                    <MSBuild Projects=`SkipNonexistentTargets.csproj` Targets=`t_nonexistent` SkipNonexistentTargets=`true` />
+                        <MSBuild Projects=`SkipNonexistentTargets.csproj` Targets=`t_nonexistent` SkipNonexistentTargets=`false` />
```

Comment:
Maybe build again with SkipNonexistentTargets=`true` to ensure it actually gets turned off.

Reply:
Oops, the other way around. Have three msbuild task invocations with false, true, false

---
In reply to: [141172470](https://github.com/Microsoft/msbuild/pull/2541#discussion_r141172470) [](ancestors = 141172470)